hardware_interface/src/resource_manager.cpp

```diff
@@ -454,6 +454,47 @@ class ResourceStorage
     hardware_info_map_[hardware.get_name()].command_interfaces = add_command_interfaces(interfaces);
   }
 
+  /// Adds exported state interfaces into internal storage.
+  /**
+   * Adds state interfaces to the internal storage. State interfaces exported from hardware or
+   * chainable controllers are moved to the map with name-interface pairs and available list's
+   * size is increased to reserve storage when interface change theirs status in real-time
+   * control loop.
+   *
+   * \param[interfaces] list of state interface to add into storage.
+   * \returns list of interface names that are added into internal storage. The output is used to
+   * avoid additional iterations to cache interface names, e.g., for initializing info structures.
+   */
```

Comment:
I think it may be better to not duplicate docstrings in the cpp. I know it's handy when developing as a reminder and also when reading but I think the header file is the place where this should be and comments in the CPP file should be reserved for implementation-specific notes if applicable

Reply:
```suggestion
```

Reply:
@bmagyar I just verified the code, I had to add it in the cpp file because these methods belong to the ResourceStorage class and this is defined inside the cpp file and there is no declaration of the methods in the header files.

What should I do now in that case?